packages/components/src/slider/slider/hook.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useMemo, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from '../styles';
+import { useContextSystem, WordPressComponentProps } from '../../ui/context';
+import { useFormGroupContextId } from '../../ui/form-group';
+import { useControlledValue, useCx } from '../../utils/hooks';
+import { interpolate } from '../../utils/interpolate';
+import { parseCSSUnitValue, createCSSUnitValue } from '../../utils/unit-values';
+import { isValueNumeric } from '../../utils/values';
+
+import type { SliderProps } from '../types';
+
+const noop = () => {};
+
+export function useSlider(
+	props: WordPressComponentProps< SliderProps, 'input', false >
+) {
+	const {
+		className,
+		defaultValue,
+		error,
+		onBlur = noop,
+		onChange: onChangeProp = noop,
+		onFocus = noop,
+		id: idProp,
+		isFocused: isFocusedProp = false,
+		max = 100,
+		min = 0,
+		size = 'default',
+		style,
+		thumbColor,
+		trackColor,
+		trackBackgroundColor,
+		value: valueProp,
+		...otherProps
+	} = useContextSystem( props, 'Slider' );
+
+	const [ _value, onChange ] = useControlledValue( {
+		defaultValue: defaultValue || '50',
+		onChange: onChangeProp,
+		value: valueProp,
+	} );
+	const [ value, initialUnit ] = parseCSSUnitValue( `${ _value }` );
+
+	const id = useFormGroupContextId( idProp );
+	const [ isFocused, setIsFocused ] = useState( isFocusedProp );
+
+	const handleOnBlur = useCallback(
+		( event ) => {
+			onBlur( event );
+			setIsFocused( false );
+		},
+		[ onBlur ]
+	);
+
+	const handleOnChange = useCallback(
+		( event ) => {
+			const nextValue = parseFloat( event.target.value );
+			if ( ! isValueNumeric( nextValue ) ) {
+				return;
+			}
+
+			let next = `${ nextValue }`;
+
+			if ( initialUnit ) {
+				next = createCSSUnitValue( nextValue, initialUnit );
+			}
+
+			onChange( next );
+		},
+		[ onChange, initialUnit ]
+	);
+
+	const handleOnFocus = useCallback(
+		( event ) => {
+			onFocus( event );
+			setIsFocused( true );
+		},
+		[ onFocus ]
+	);
+
+	const currentValue = interpolate(
```

Comment:
It looks like a better name for this variable could be `progressPerc`
```suggestion
	// Interpolate the current value between 0 and 100, so that it can be used to
	// position the slider's thumb correctly
	const progressPerc = interpolate(
```

Reply:
I've tweaked the naming of this variable to be more fit for purpose and included the explanatory comment. See https://github.com/WordPress/gutenberg/pull/42315/commits/47ec300c1bf95fa6274cae54c211ba4f0b7acaa3.